resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v1/Client.scala

```diff
@@ -554,7 +556,7 @@ private[spark] class Client(
   private class DriverServiceReadyWatcher(resolvedDriverService: SettableFuture[Service])
       extends Watcher[Service] {
     override def eventReceived(action: Action, service: Service): Unit = {
-      if ((action == Action.ADDED) || (action == Action.MODIFIED)
+      if ((action == Action.ADDED || action == Action.MODIFIED)
```

Comment:
the parentheses change slightly modifies the evaluation order of these expressions, and the new ordering seems correct.

Java precedence for boolean expressions is `!` then `&&` then `||` so previously this was evaluating the `&&` first and then the `||`